### DISCUSSION TITLE
Can Optuna TPE constraints function can accept parameters?

### DISCUSSION THREAD

**Discussion by okaikov** [optuna/optuna#4643] [Q&A]
Hello,
In my use case I want to change my constraints function in the middle of the optimization.
Is it possible to provide a parameter to the constraints function? Example: constraints_func(trial=trial, limits=limits)
In the examples provided in the Optuna website, the constraints function is reading hardcoded values from the user_attrs.
Since I'm changing the limits, I want to be able to provide the function the params to calculate the new limits.

**Comment by nzw0301** [comment]
Similar to objective function, you can try https://optuna.readthedocs.io/en/stable/faq.html#how-to-define-objective-functions-that-have-own-arguments.
